File: wallet_syncronizer/pkg/util/json/json.go
package json

import "github.com/gofiber/fiber/v2"

type Response struct {
	Error *fiber.Error `json:"error,omitempty"`
	Data  interface{}  `json:"data,omitempty"`
}

func NewErrorResponse(httpStatus int, message string) Response {
	return Response{
		Error: fiber.NewError(httpStatus, message),
	}
}

File: wallet_syncronizer_node/util/json.js
class Response {
    constructor(error, data) {
        this.error = error;
        this.data = data;
    }
}

function NewErrorResponse(httpStatus, message) {
    return new Response({
        status: httpStatus,
        message: message,
    }, null);
}

function NewSuccessResponse(data) {
    return new Response(null, data);
}

module.exports = {
    Response,
    NewErrorResponse,
    NewSuccessResponse,
};
